packages/experimental-utils/src/ts-eslint/Rule.ts

```diff
@@ -32,6 +32,11 @@ interface RuleMetaDataDocs {
    * The URL of the rule's docs
    */
   url: string;
+  /**
+   * Does the rule require us to create a full TypeScript Program in order for it
+   * to type-check code
```

Comment:
Because this is exported as our "public api" - maybe it's good to clarify that this is only for documentation, so people don't think it's required to use type information
WDYT?

```suggestion
   * to type-check code. This is only used for documentation purposed.
```